index.js

```diff
@@ -23,17 +23,22 @@ function addLayers(geojson, sourceId, layers) {
             switch (geojson.geometry.type) {
                 case 'Point':
                     if (!geojson.properties) geojson.properties = {};
-                    geojson.properties._id = hat();
+                    geojson.properties._id = cuid();
                     layers.push(makeLayer(geojson, sourceId, 'Point'));
                     break;
                 case 'LineString':
                     if (!geojson.properties) geojson.properties = {};
-                    geojson.properties._id = hat();
+                    geojson.properties._id = cuid();
                     layers.push(makeLayer(geojson, sourceId, 'LineString'));
+
+                    // Encoded polylines can be rendered as polgons. If a LineString has fill properties, add them.
+                    if ('fill' in geojson.properties || 'fill-opacity' in geojson.properties) {
+                        layers.push(makeLayer(geojson, sourceId, 'Polygon'));
```

Comment:
If I’m reading this correctly, the fill part of the overlay gets added as a fill layer after (above) the stroke part of the overlay, correct? In the case of a translucent fill, wouldn’t that cause half of the stroke to be visible through the fill? That would be aesthetically unpleasing if the stroke is opaque and thick. I’d expect the fill to be added before (below) the stroke to avoid this issue. (I suppose this is also true of the `Polygon` case below.)